File: livingrimoire start here/LivinGrimoire java/src/Auxiliary_Modules/LGFIFO.java
package Auxiliary_Modules;

import java.util.ArrayList;
import java.util.Iterator;
import java.util.Random;

public class LGFIFO <T>{
    //first in first out queue
    protected ArrayList<T> elements = new ArrayList<>();
    public void add(T item){
        elements.add(item);
    }
    public int size(){
        return elements.size();
    }
    public T peak(){
        if(size() == 0){
            return null;
        }
        return elements.get(0);
    }
    public T poll(){
        if(size() == 0){
            return null;
        }
        T result = elements.get(0);
        elements.remove(0);
        return result;
    }
    public void removeItem(T item){
        elements.remove(item);
    }
    public void clear(){
        elements.clear();
    }
    private Random rand = new Random();
    public T getRNDElement(){
        if (elements.isEmpty()) {
            return null;
        }
        return  elements.get(rand.nextInt(elements.size()));
    }
    public boolean contains(T input){
        return elements.contains(input);
    }
    public Boolean isEmpty(){
        return elements.isEmpty();
    }
    public void remove(T element){
        elements.remove(element);
    }
    public Iterator<T> iterator(){
        return elements.iterator();
    }

    public ArrayList<T> getElements() {
        return elements;
    }
    public void setElements(ArrayList<T> elements) {
        this.elements = elements;
    }
}

File: livingrimoire start here/LivinGrimoire Kotlin/kotlin/auxiliary_modules/LGFIFO.kt
package auxiliary_modules

import java.util.*

open class LGFIFO<T> {
    //first in first out queue
    var elements = ArrayList<T>()

    open fun add(item: T) {
        elements.add(item)
    }

    fun size(): Int {
        return elements.size
    }

    open fun peak(): T? {
        return if (size() == 0) {
            null
        } else elements[0]
    }

    open fun poll(): T? {
        if (size() == 0) {
            return null
        }
        val result = elements[0]
        elements.removeAt(0)
        return result
    }

    open fun removeItem(item: T) {
        if (elements.contains(item)) {
            elements.remove(item)
        }
    }

    fun clear() {
        elements.clear()
    }

    private val rand = Random()
    open val rNDElement: T?
        get() = if (elements.isEmpty()) {
            null
        } else elements[rand.nextInt(elements.size)]

    operator fun contains(input: T): Boolean {
        return elements.contains(input)
    }

    val isEmpty: Boolean
        get() = elements.isEmpty()

    fun remove(element: T) {
        elements.remove(element)
    }

    operator fun iterator(): Iterator<T> {
        return elements.iterator()
    }
}